examples/4-exchanges/src/index.re

```diff
@@ -0,0 +1,78 @@
+open ReasonUrql;
+
+/* This is the native debugExchange that ships with `urql`, re-implemented in Reason.
+     Typically, you'd just add Exchanges.debugExchange to the Client's exchange array.
+   */
+let debugExchange: Exchanges.exchange =
+  exchangeInput => {
+    let forward = Exchanges.forwardGet(exchangeInput);
+
+    ops =>
+      ops
+      |> Wonka.tap((. op) =>
+           Js.log2("[Exchange debug]: Incoming operation: ", op)
+         )
+      |> forward
+      |> Wonka.tap((. res) =>
+           Js.log2("[Exchange debug]: Completed operation: ", res)
+         );
+  };
+
+let client =
+  Client.make(
+    ~url="https://formidadog-ql.now.sh",
+    ~exchanges=[|debugExchange, Exchanges.fetchExchange|],
+    (),
+  );
+
+module GetAllDogs = [%graphql
+  {|
+  query dogs {
+    dogs {
+      key
+      name
+      breed
+      likes
+    }
+  }
+|}
+];
+
+let queryRequest = Request.createRequest(~query=GetAllDogs.make()##query, ());
+
+module LikeDog = [%graphql
+  {|
+  mutation likeDog($key: ID!) {
+    likeDog(key: $key) {
+      name
+      likes
+    }
+  }
+  |}
+];
+
+Client.executeQuery(~client, ~query=queryRequest)
+|> Wonka.subscribe((. response) => {
```

Comment:
I'll flesh this example out more a bit later on. Right now, it just kicks off a mutation every 5 seconds so you can see the `debugExchange` working in the console.